python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py

```diff
@@ -160,13 +209,81 @@ async def _inner_get_streaming_chat_message_contents(
 
         async for part in response_object:
             yield [
-                StreamingChatMessageContent(
-                    role=AuthorRole.ASSISTANT,
-                    choice_index=0,
-                    inner_content=part,
-                    ai_model_id=self.ai_model_id,
-                    content=part.get("message", {"content": None}).get("content", None),
+                self._create_streaming_chat_message_content(
+                    part,
+                    self._get_meta_data_from_response(part),
                 )
             ]
 
     # endregion
+
+    def _create_chat_message_content(self, response: Mapping[str, Any], metadata: dict[str, Any]) -> ChatMessageContent:
+        """Create a chat message content from the response."""
+        items: list[ITEM_TYPES] = []
+        if not (message := response.get("message", None)):
+            raise ServiceInvalidResponseError("No message content found in response.")
+
+        if content := message.get("content", None):
+            items.append(
+                TextContent(
+                    text=content,
+                    inner_content=message,
+                )
+            )
+        if tool_calls := message.get("tool_calls", None):
+            for tool_call in tool_calls:
+                items.append(
+                    FunctionCallContent(
+                        inner_content=tool_call,
+                        ai_model_id=self.ai_model_id,
+                        name=tool_call.get("function").get("name"),
+                        arguments=tool_call.get("function").get("arguments"),
+                    )
+                )
+
+        return ChatMessageContent(
+            role=AuthorRole.ASSISTANT,
+            items=items,
+            inner_content=response,
+            metadata=metadata,
+        )
+
+    def _create_streaming_chat_message_content(
+        self, part: Mapping[str, Any], metadata: dict[str, Any]
+    ) -> StreamingChatMessageContent:
+        """Create a streaming chat message content from the response part."""
+        items: list[STREAMING_ITEM_TYPES] = []
+        if not (message := part.get("message", None)):
+            raise ServiceInvalidResponseError("No message content found in response part.")
+
+        if content := message.get("content", None):
+            items.append(
+                StreamingTextContent(
+                    choice_index=0,
+                    text=content,
+                    inner_content=message,
+                )
+            )
+
+        return StreamingChatMessageContent(
+            role=AuthorRole.ASSISTANT,
+            choice_index=0,
+            items=items,
+            inner_content=part,
+            ai_model_id=self.ai_model_id,
+            metadata=metadata,
+        )
+
+    def _get_meta_data_from_response(self, response: Mapping[str, Any]) -> dict[str, Any]:
```

Comment:
This is a super-nit, so please feel free to disregard, but is there a reason we are spelling `metadata` as `meta_data`?

Reply:
Not at all. This is super valid. My mistake. Corrected.